src/hooks/useSnapshotDelegateContract.ts

```diff
@@ -0,0 +1,38 @@
+import { hexlify } from '@ethersproject/bytes'
+import { Contract } from '@ethersproject/contracts'
+import { Web3Provider } from '@ethersproject/providers'
+import useAuth from 'decentraland-gatsby/dist/hooks/useAuth'
+import { useMemo } from 'react'
+import { SNAPSHOT_SPACE } from '../entities/Snapshot/constants'
+import DelegateABI from '../modules/contracts/abi/Delegate.abi.json'
+
+function useSnapshotDelegateContract() {
+
+  const [userAddress, authState] = useAuth()
+
+  return useMemo(() => {
+
+    let isContractUsable = false
+    let setDelegate = null
+    let clearDelegate = null
+    let checkDelegation = null
+
+    if (authState.provider) {
+      const enc = new TextEncoder()
+      const signer = new Web3Provider(authState.provider).getSigner()
```

Comment:
I think you can write this `const signer = authState.provider ? new Web3Provider(authState.provider).getSigner() : undefined`

Reply:
And remove `authState.provider` if condition

Reply:
I'm not sure of this approach because later you'd need to check if signer is undefined or not...

Maybe we can write it like 

```typescript
  let setDelegate = null
  let clearDelegate = null
  let checkDelegation = null
  
  if(authState.provider && contractAddress){
    // ...
   }
   
   const contractIsUsable = setDelegate && clearDelegate && checkDelegation
   
   return { contractIsUsable, setDelegate, clearDelegate, checkDelegation }
```

Reply:
Maybe we could even avoid the `isContractUsable` variable, and just do
`return {setDelegate, clearDelegate, checkDelegation }` since we already need to ask each time we want to use one of the contract functions.

We can do it like 

`isContractUsable && clearDelegate! ` => `clearDelegate && clearDelegate()`
`isContractUsable && setDelegate!(address)` => `setDelegate && setDelegate(address)`

```typescript
if (isContractUsable) {
        return await checkDelegation!()
      }
      return null
    },
```
=> `return checkDelegation && await checkDelegation()` ?

Reply:
I wouldn't initialize the functions as null either... I'd rather check "if contract is usable" inside of each function. But always expect a function from the component using the hook.